### PR TITLE
Adding OBO test when Scope changes and cache is empty

### PR DESCRIPTION
**Pull request by trwalke**
Fixes # 
continuation from here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3266

**Changes proposed in this request**
Adding tests
code clean up

**Testing**
Unit

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs [hunk 1]
```diff
@@ -480,6 +480,56 @@ public async Task AcquireTokenByObo_NormalOboThenLongRunningInitiate_WithTheSame
...
+
+            var result = await cca.InitiateLongRunningProcessInWebApi(s_scopes, userAuthResult.AccessToken, ref oboCacheKey)
+                .ExecuteAsync().ConfigureAwait(false);
+
+            // Cache has 1 partition (user-provided key) with 1 token
+            Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+            Assert.AreEqual(1, cca.UserTokenCacheInternal.Accessor.GetAllAccessTokens().Count);
+            Assert.AreEqual(1, cca.UserTokenCacheInternal.Accessor.GetAllRefreshTokens().Count);
+
+            // Returns long-running token
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs:512]:
```suggestion
            // No matching AT, uses RT to retrieve new AT.
```

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs [hunk 2]
```diff
@@ -480,6 +480,56 @@ public async Task AcquireTokenByObo_NormalOboThenLongRunningInitiate_WithTheSame
             Assert.AreEqual(1, cca.UserTokenCacheInternal.Accessor.GetAllRefreshTokens().Count);
         }
 
+        [TestMethod]
+        public async Task AcquireTokenByObo_LongRunning_WithDifferentScopes_TestAsync()
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs:484]: Wondering if we should add another test method, similar to this one but when there's no AT in the cache at all, since we have that slight branching when we check if GetAllAT returns 0 tokens? Maybe a unit test instead of integration?

**Review by pmaytak — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs [hunk 3]
```diff
@@ -480,6 +480,56 @@ public async Task AcquireTokenByObo_NormalOboThenLongRunningInitiate_WithTheSame
...
+            string oboCacheKey = "obo-cache-key";
+            UserAssertion userAssertion = new UserAssertion(userAuthResult.AccessToken);
+
+            var result = await cca.InitiateLongRunningProcessInWebApi(s_scopes, userAuthResult.AccessToken, ref oboCacheKey)
+                .ExecuteAsync().ConfigureAwait(false);
+
+            // Cache has 1 partition (user-provided key) with 1 token
+            Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+            Assert.AreEqual(1, cca.UserTokenCacheInternal.Accessor.GetAllAccessTokens().Count);
+            Assert.AreEqual(1, cca.UserTokenCacheInternal.Accessor.GetAllRefreshTokens().Count);
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs:510]: nit: add some scope assertions and more comments around how this works

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs [hunk 4]
```diff
@@ -480,6 +480,56 @@ public async Task AcquireTokenByObo_NormalOboThenLongRunningInitiate_WithTheSame
...
+
+            // Returns long-running token
+            result = await cca.AcquireTokenInLongRunningProcess(scopes2, oboCacheKey).ExecuteAsync().ConfigureAwait(false);
+
+            Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs:515]: The IdentityProvider should be IDP? Do we have a bug here?

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
Looks like there is a bug because when you perform the refresh_token flow the TokenSource is set to Cache, instead of IdP.

**Review by bgavrilMS — APPROVED** [review]
